model-sync-gradle-test/graph-lang-api/build.gradle.kts

```diff
@@ -0,0 +1,68 @@
+plugins {
+    id("org.modelix.model-api-gen")
+    `maven-publish`
+    kotlin("jvm") version "1.9.0"
+}
+
+repositories {
+    mavenLocal()
+    gradlePluginPortal()
+    maven { url = uri("https://artifacts.itemis.cloud/repository/maven-mps/") }
+    mavenCentral()
+}
+
+val modelixCoreVersion = file("../../version.txt").readText()
+
+version = modelixCoreVersion
+
+val mps by configurations.creating
+val kotlinGenDir = buildDir.resolve("metamodel/kotlin").apply { mkdirs() }
+
+dependencies {
+    mps("com.jetbrains:mps:2021.1.4")
+    api("org.modelix:model-api-gen-runtime:$modelixCoreVersion")
+}
+
+val copyGeneratedApiToSrc by tasks.registering(Sync::class) {
+    dependsOn(tasks.named("generateMetaModelSources"))
+    from(kotlinGenDir)
+    into("src/main/kotlin")
+}
```

Comment:
Why not modifying the sourceSet instead of copying files? I would avoid putting generated files into the actual source tree. That makes it easy to accidentally commit them.

Reply:
The generated api is the only source code for that project a separate source set is not necessary. Accidentally committing files should be quite hard, as this src dir is part of the .gitignore file.